src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/InstanceEntryPointTableNode.cs

```diff
@@ -0,0 +1,166 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection.Metadata.Ecma335;
+
+using Internal.NativeFormat;
+using Internal.Runtime;
+using Internal.Text;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+
+namespace ILCompiler.DependencyAnalysis.ReadyToRun
+{
+    public class InstanceEntryPointTableNode : HeaderTableNode
+    {
+        private struct EntryPoint
+        {
+            public static EntryPoint Null = new EntryPoint(-1, -1, -1, 0);
+            
+            public readonly int MethodIndex;
+            public readonly int FixupIndex;
+            public readonly int SignatureIndex;
+            public readonly int MethodHashCode;
+
+            public bool IsNull => (MethodIndex < 0);
+            
+            public EntryPoint(int methodIndex, int fixupIndex, int signatureIndex, int methodHashCode)
+            {
+                MethodIndex = methodIndex;
+                FixupIndex = fixupIndex;
+                SignatureIndex = signatureIndex;
+                MethodHashCode = methodHashCode;
+            }
+        }
+
+        List<EntryPoint> _ridToEntryPoint;
+
+        List<byte[]> _uniqueFixups;
+        Dictionary<byte[], int> _uniqueFixupIndex;
+
+        List<byte[]> _uniqueSignatures;
+        Dictionary<byte[], int> _uniqueSignatureIndex;
+        
+        public InstanceEntryPointTableNode(TargetDetails target)
+            : base(target)
+        {
+            _ridToEntryPoint = new List<EntryPoint>();
+
+            _uniqueFixups = new List<byte[]>();
+            _uniqueFixupIndex = new Dictionary<byte[], int>(ByteArrayComparer.Instance);
+
+            _uniqueSignatures = new List<byte[]>();
+            _uniqueSignatureIndex = new Dictionary<byte[], int>(ByteArrayComparer.Instance);
+        }
+        
+        public override void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
+        {
+            sb.Append(nameMangler.CompilationUnitPrefix);
+            sb.Append("__ReadyToRunInstanceEntryPointTable");
+        }
+
+        public void Add(MethodCodeNode methodNode, int methodIndex)
+        {
```

Comment:
Hmm, this is wrong, will fix - we're not really differentiating non-generic and generic methods in the two classes.

Reply:
Fixed in 6th commit.